yamllint/plugins.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Satoru SATOH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Plugin module utilizing setuptools (pkg_resources) to allow users to add their
+own custom lint rules.
+"""
+try:
+    from pkg_resources import iter_entry_points
+except ImportError:
+    def iter_entry_points(group):
+        """Dummy function for pkg_resources.iter_entry_points."""
+        yield
+
+
+PACKAGE_GROUP = "yamllint.plugins"
+
+
+def validate_rule_module(rule_mod):
+    """Test if given rule module is valid.
+    """
+    return getattr(rule_mod, "ID", False
+                   ) and callable(getattr(rule_mod, "check", False))
+
+
+def load_plugin_rules_itr(group=PACKAGE_GROUP):
+    """Load custom lint rule plugins."""
+    for entry in iter_entry_points(group):
+        try:
+            rules_mod = entry.load()
+            yield rules_mod.RULES_MAP
+        except AttributeError:
+            pass  # TBD
```

Comment:
What do you mean with `TBD`?

What do you think about removing `load_plugin_rules_itr()` and loading plugins into `update_rules_map()`, without `try`/`except`? If importing a plugin fails → no silent error. The problem is clearly visible in the stacktrace and the user knows *why* the custom plugin doesn't work, and *how* to fix it (uninstall the plugin or fix it).

Reply:
TBD means to be determined to fix what to do if plugin is broken or has some problems to work.
I have some ideas about it, e.g. output warning messages about broken plugins, but I'm not sure twhat is the best to do for such cases you think so that I commented as it is.

I think that maybe there are cases that the quality of plugin code is not enough compared to yamllint itself you keep maintained and it would be better to keep yamllint works even if such plugins are installed.